### PR TITLE
Feature/iss 15 change join to use arel

### PR DESCRIPTION
**Pull request by midwire**
Addresses issue: https://github.com/midwire/my_zipcode_gem/issues/15

### CODE REVIEW

File: lib/generators/my_zipcode_gem/templates/county_model.rb [hunk 1]
```diff
@@ -8,8 +8,16 @@ class County < ActiveRecord::Base
   validates :name, uniqueness: { scope: :state_id, case_sensitive: false },
                    presence: true
 
-  scope :without_zipcodes, -> { joins("LEFT JOIN zipcodes ON zipcodes.county_id = counties.id").where("zipcodes.county_id IS NULL") }
-  scope :without_state, -> { where("state_id IS NULL") }
+  scope :without_zipcodes, -> {
+    county = County.arel_table
+    zipcodes = Zipcode.arel_table
+    zipjoin = county.join(zipcodes, Arel::Nodes::OuterJoin)
+        .on(zipcodes[:county_id].eq(county[:id]))
```
Comment (midwire) [inline lib/generators/my_zipcode_gem/templates/county_model.rb]: I don't remember ever telling hound to review this code :).  Maybe you did which is absolutely fine. Maybe we can update the rules to follow what we want.

Comment (cmckni3) [inline lib/generators/my_zipcode_gem/templates/county_model.rb]: I think it has been setup as a hook. I agree with aligning `.on` with `.join`

File: lib/generators/my_zipcode_gem/templates/county_model.rb [hunk 2]
```diff
@@ -8,8 +8,16 @@ class County < ActiveRecord::Base
...
+    joins(zipjoin.join_sources).where(zipcodes[:county_id].eq(nil))
+  }
+  scope :without_state, -> {
```
Comment (cmckni3) [inline lib/generators/my_zipcode_gem/templates/county_model.rb]: I agree with this one too

File: README.md [hunk 3]
```diff
@@ -13,25 +13,30 @@ Simple gem to handle zipcode lookups and related functionality.  `rake zipcodes:
 
 Add the following line to your Gemfile:
 
-    gem 'my_zipcode_gem'
```
Comment (cmckni3) [inline README.md]: 👍

**Comment by cmckni3** [comment]
I would make those small style changes at some point. LGTM :shipit:

File: lib/generators/my_zipcode_gem/templates/county_model.rb [hunk 4]
```diff
@@ -8,8 +8,17 @@ class County < ActiveRecord::Base
...
                    presence: true
 
-  scope :without_zipcodes, -> { joins("LEFT JOIN zipcodes ON zipcodes.county_id = counties.id").where("zipcodes.county_id IS NULL") }
-  scope :without_state, -> { where("state_id IS NULL") }
+  scope :without_zipcodes, lambda {
+    county = County.arel_table
+    zipcodes = Zipcode.arel_table
+    zipjoin = county
+        .join(zipcodes, Arel::Nodes::OuterJoin)
+        .on(zipcodes[:county_id].eq(county[:id]))
```
Comment (midwire) [inline lib/generators/my_zipcode_gem/templates/county_model.rb]: Yeah, I'm not sure I agree with this either. Not sure how to turn that one off.

Comment (cmckni3) [inline lib/generators/my_zipcode_gem/templates/county_model.rb]: Looks like `Style/DotPosition` is the rule. It looks like leading is the default in rubocop though.

Maybe Hound overrides it and it needs to be manually set.

```
# Multi-line method chaining should be done with leading dots.
Style/DotPosition:
  EnforcedStyle: leading
```

**Pull request merged**